ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java

```diff
@@ -171,6 +192,26 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
             Optional.empty()));
   }
 
+  private boolean validateWithdrawals(final EnginePayloadAttributesParameter payloadAttributes) {
+    // TODO Withdrawals move this into mergeCoordinator.updateForkChoice with the other
+    // payloadAttributes validation?
+    final List<Withdrawal> withdrawals =
+        Optional.ofNullable(payloadAttributes.getWithdrawals())
+            .map(ws -> ws.stream().map(WithdrawalParameter::toWithdrawal).collect(toList()))
+            .orElse(null);
+    return timestampSchedule
+        .getByTimestamp(payloadAttributes.getTimestamp())
+        .map(
+            protocolSpec -> protocolSpec.getWithdrawalsValidator().validateWithdrawals(withdrawals))
```

Comment:
Is it necessary to have this on the protocol schedule? Do V2 APIs need to work before the shanghai?

Reply:
Yes, some CLs e.g. lighthouse always send V2

Reply:
Could the V2 APIs use the allowed withdrawals validator and the V1 APIs use the not allowed validator instead of driving this from the protocol schedule? Doesn't look like ProtocolSpec needs a getWithdrawalsValidator method.

Reply:
I believe the correct interpretation of the spec is the validation is driven by Shanghai, not by the method name. 

Currently, driving it from V2 would break lighthouse because we'd invalidate pre-shanghai blocks, see https://discord.com/channels/595666850260713488/892088344438255616/1055785287885783040. 

Once shanghai has passed, we'd also be relying on the CLs to send V1 for pre-shanghai blocks (e.g. in syncing scenarios) which implies clients need to keep all versions around forever, hence the aim of backwards compatibility.